src/core/thread/mesh_forwarder.cpp

```diff
@@ -1014,6 +1033,18 @@ void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame)
         {
             child->mFragmentOffset = 0;
             mSendMessage->ClearChildMask(mMle.GetChildIndex(*child));
+
+            if ((child->mMode & Mle::ModeTlv::kModeRxOnWhenIdle) == 0)
+            {
+                mMac.ClearSrcMatchAddressEntry(macDest);
+
```

Comment:
Agree.  We need to consider the case where we have multiple messages pending for a sleepy child.